Source/WebCore/html/parser/HTMLParserScheduler.cpp

```diff
@@ -101,6 +101,16 @@ void HTMLParserScheduler::continueNextChunkTimerFired()
     m_parser.resumeParsingAfterYield();
 }
 
+static bool parsingProgressedSinceLastYield(PumpSession& session)
+{
+    // Only yield if there has been progress since last yield.
+    if (session.processedTokens > session.processedTokensOnLastYieldBeforeScript) {
+        session.processedTokensOnLastYieldBeforeScript = session.processedTokens;
```

Comment:
The name of the variable `session.processedTokensOnLastYieldBeforeScript` is not super accurate since we only update it if we get here.

Reply:
Ah, yes, that is a good point. Since the name is already quite long, I wonder if we should look into calling this function in the other places where we yield before executing a script, since we should not yield if there hasn't been any progress on parsing. I will look into that in a follow-up patch.

Thanks for reviewing!